pkgs/development/tools/tic80/default.nix

```
@@ -0,0 +1,60 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, ruby
+, xcbuild
+, AVFoundation
+, AudioUnit
+, Cocoa
+, CoreAudio
+, CoreServices
+, ForceFeedback
+, OpenGL
+}:
+
+stdenv.mkDerivation rec {
+  pname = "tic-80";
+
+  version = "1.0.2164";
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  buildInputs = [
+    # Ruby support!
+    ruby
+  ] ++ lib.optional stdenv.isDarwin ([
+    xcbuild
+  ] ++ [
+    AVFoundation
+    AudioUnit
+    Cocoa
+    CoreAudio
+    CoreServices
+    ForceFeedback
+    OpenGL
+  ]);
```

It's probably should look more like this:
```suggestion
  nativeBuildInputs = [
    cmake
    ruby
  ] ++ lib.optionals stdenv.isDarwin [
    xcbuild
  ];

  buildInputs = lib.optionals stdenv.isDarwin [
    AVFoundation
    AudioUnit
    Cocoa
    CoreAudio
    CoreServices
    ForceFeedback
    OpenGL
  ];
```